onnxruntime/core/mlas/lib/halfgemm.h

```diff
@@ -513,3 +515,5 @@ MlasHalfGemmGetDispatch()
     return &MlasHalfGemmDispatchDefault;
 #endif
 }
+
+#endif
```

Comment:
general: it's helpful to have a comment with the condition on the `#endif` especially when the `#if` is further away

```suggestion
#endif  // !defined(ORT_MINIMAL_BUILD) || defined(ORT_EXTENDED_MINIMAL_BUILD)
```